workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/WorkflowUiViewState.kt

```diff
@@ -0,0 +1,192 @@
+    override val showing: RenderingT,
+    override val environment: ViewEnvironment,
+    override val showRendering: ViewShowRendering<RenderingT>,
+
+    val starter: (View) -> Unit = { view ->
+      view.showRendering(view.getRendering()!!, view.environment!!)
+    }
+  ) : WorkflowUiViewState<RenderingT>()
+
+  /** [start] has been called. It's safe to call [showRendering] now. */
+  data class Started<out RenderingT : Any>(
+    override val showing: RenderingT,
+    override val environment: ViewEnvironment,
+    override val showRendering: ViewShowRendering<RenderingT>
+  ) : WorkflowUiViewState<RenderingT>()
+}
+
+/**
+ * Intended for use by implementations of [ViewFactory.buildView].
+ *
+ * Establishes [showRendering] as the implementation of [View.showRendering]
+ * for the receiver, possibly replacing the existing one. Likewise sets / updates
+ * the values returned by [View.getRendering] and [View.environment].
+ *
+ * - [View.start] must be called exactly once before [View.showRendering].
+ *
+ * @throws IllegalStateException when called after [View.start]
+ *
+ * @see DecorativeViewFactory
+ */
+@WorkflowUiExperimentalApi
+public fun <RenderingT : Any> View.bindShowRendering(
+  initialRendering: RenderingT,
+  initialViewEnvironment: ViewEnvironment,
+  showRendering: ViewShowRendering<RenderingT>
+) {
+  workflowTag = when (workflowTagOrNull) {
+    is New<*> -> New(initialRendering, initialViewEnvironment, showRendering, starter)
+    else -> New(initialRendering, initialViewEnvironment, showRendering)
+  }
+}
+
+/**
+ * It is usually more convenient to use [WorkflowViewStub] than to call this method directly.
+ *
+ * Makes the initial call to [View.showRendering], along with any wrappers that have been
+ * added via [ViewRegistry.buildView], or [DecorativeViewFactory.viewStarter].
+ *
+ * - It is an error to call this method more than once.
+ * - It is an error to call [View.showRendering] without having called this method first.
+ */
+@WorkflowUiExperimentalApi
+public fun View.start() {
+  val current = workflowTagAsNew
+  workflowTag = Started(current.showing, current.environment, current.showRendering)
+  current.starter(this)
+}
+
+/**
+ * It is usually more convenient to use [WorkflowViewStub] than to call this method directly.
+ *
+ * True if this view is able to show [rendering].
+ *
+ * Returns `false` if [View.bindShowRendering] has not been called, so it is always safe to
+ * call this method. Otherwise returns the [compatibility][compatible] of the current
+ * [View.getRendering] and the new one.
+ */
+@WorkflowUiExperimentalApi
+public fun View.canShowRendering(rendering: Any): Boolean {
+  return getRendering<Any>()?.matches(rendering) == true
+}
+
+/**
+ * It is usually more convenient to use [WorkflowViewStub] than to call this method directly.
+ *
+ * Sets the rendering associated with this view, and displays it by invoking
+ * the [ViewShowRendering] function previously set by [bindShowRendering].
```

Comment:
```suggestion
 * Shows [rendering] in this View by invoking
 * the [ViewShowRendering] function previously set by [bindShowRendering].
```

I get its setting a tag but that is the implementation detail I think and not necessary to the outcome of the function.